src/main/java/io/cdap/e2e/pages/locators/CdfNameSpaceAdminLocators.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.e2e.pages.locators;
+
+import io.cdap.e2e.utils.SeleniumDriver;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+/**
+ * Represents CdfNameSpaceAdminLocators
+ */
+public class CdfNameSpaceAdminLocators {
+
+  @FindBy(how = How.XPATH, using = "//div[contains(text(),'Namespace Admin')]")
+  public static WebElement namespaceAdminMenu;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='navbar-hamburger-icon']")
+  public static WebElement hamburgerMenu;
+
+  public static WebElement nameSpaceModules(String module, String nameSpace) {
+    String path = "//*[@href=\"/cdap/ns/" + nameSpace + "/details/" + module + "\"]";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  public static WebElement createProfile(String nameSpace) {
+    String path = "//*[@href=\"/cdap/ns/" + nameSpace + "/profiles/create\"]";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  public static WebElement switchNameSpace(String nameSpace) {
+    String path = "//*[@href=\"/cdap/ns/'" + nameSpace + "'\"]";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  public static WebElement locateMenuLink(String menuLink) {
+    String xpath = "//*[@data-cy='navbar-" + menuLink + "-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateKeyProperty(String keyProperty, int row) {
+    String xpath = "//*[@data-cy='" + keyProperty + "" + row + "']//input[@placeholder='key']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateValueProperty(String valueProperty, int row) {
+    String xpath = "//*[@data-cy='" + valueProperty + "" + row + "']//input[@placeholder='value']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateAddRowButtonProperty(String addRowProperty, int row) {
+    String xpath =
+        "//*[@data-cy='" + addRowProperty + "" + row + "']//span"
+            + "//button[@type='submit' and @class='btn add-row-btn btn-link']";
```

Comment:
locating elements using class names is not a good idea, as it prevents the UI developers from changing the design / implementation of the specific UI elements. Prefer `data-testid`s for locating elements.